src/controllers/queryRunner.ts

```diff
@@ -380,38 +381,81 @@ export default class QueryRunner {
             }
         }
 
+        // add the column headers
+        if (this.shouldIncludeHeaders(includeHeaders)) {
+            let firstCol: number;
+            let lastCol: number;
+            for (let range of selection) {
+                if (firstCol === undefined || (range.fromCell < firstCol)) {
+                    firstCol = range.fromCell;
+                }
+                if (lastCol === undefined || (range.toCell > lastCol)) {
+                    lastCol = range.toCell;
+                }
+            }
+            let columnRange: ISlickRange = {
+                fromCell: firstCol,
+                toCell: lastCol,
+                fromRow: undefined,
+                toRow: undefined
+            };
+            let columnHeaders = this.getColumnHeaders(batchId, resultId, columnRange);
+            copyString += columnHeaders.join('\t');
+        }
+        copyString += os.EOL;
```

Comment:
You should also be trimming off the last EOL

Reply:
why? we want the column headers to be in separate line

Reply:
Sorry, missed that this was for the header row only.

Let's make sure that if there's no data we're still trimming off this last EOL though to be consistent.